src/deadline/job_attachments/asset_sync.py

```diff
@@ -337,6 +338,19 @@ def _record_attachment_mtimes(
                 abs_path = str(Path(local_root) / manifest_path.path)
                 self.synced_assets_mtime[abs_path] = Path(abs_path).stat().st_mtime_ns
 
+    def _ensure_disk_capacity(self, session_dir: Path, total_input_bytes: int) -> None:
+        """
+        Raises an AssetSyncError if the given input bytes is larger than the available disk space.
+        """
+        disk_free: int = shutil.disk_usage(session_dir).free
+        if total_input_bytes > disk_free:
+            input_size_readable = _human_readable_file_size(total_input_bytes)
+            disk_free_readable = _human_readable_file_size(disk_free)
+            raise AssetSyncError(
+                "Error occurred while attempting to sync input files: "
+                f"Total file size required for download ({input_size_readable}) is larger than available disk space ({disk_free_readable})"
```

Comment:
What're your thoughts on adding more guidance for the user? From what I understand they have 2 options available to them:
1. Increase storage capacity on the workers
2. Switch to VFS if they believe each task COULD fit within the storage capacity

Can you think of a succinct way to make that message user-friendly? Right now, I'm not certain how users find out that VFS is an option

Reply:
We do have a whole docs page for VFS, so customers should have ways to find out about it: https://docs.aws.amazon.com/deadline-cloud/latest/userguide/storage-virtual.html, I also put a link to this same page in the JA readme

I could link that in the code, but not sure if we've done that anywhere / if that's a thing we want to do?

Reply:
I'm struggling to come up with a succinct way to word this as well. As I don't believe customers can increase the storage of SMF workers, so point 1 doesn't help in all cases. So really we have:

1. If SMF, try VFS if a total step's input size < the total storage capacity
2. Otherwise if SMF, just make smaller jobs, or try your own CMF with larger storage
3. If CMF, increase your storage

I don't think the worker knows if it's SMF or CMF in this context, so I'd have to do a rather large message, how does this sound?

```
Error occurred while attempting to sync input files:
Total file size required for download (12PB) is larger than available disk space (155GB).
Try using the 'VIRTUAL' job attachments filesystem option if the total input size for each step is less than the available space on the worker.
Otherwise if using a CMF worker, try increasing the storage capacity.
```

Reply:
Chatted offline, I think we can do better to guide customers, but for now we'll leave this as-is. The info probably belongs in a troubleshooting section due to the decision tree that presents itself